spec/14-classes.md

```diff
@@ -396,19 +404,19 @@ Widget::__callStatic('sMethod', array(NULL, 1.234))
 
 **Constraints:**
 
-A *const-declaration* must only appear at the top level of a script, be
-a *class constant* (inside a [*class-definition*](#class-members) or be an
-[*interface constant*](15-interfaces.md#interface-members).
+A *const-declaration* must only appear at the top level of a script, and
+must not redefine an existing [c-constant](06-constants.md#general).
 
-A *const-declaration* must not redefine an existing [c-constant](06-constants.md#general).
+A *class-const-declaration* must be inside a [*class-definition*](#class-members) or be an
+[*interface constant*](15-interfaces.md#interface-members).
 
-A class constant must not have visibility specifier or `static` specifier.
+A class constant must not have `static` specifier.
 
 **Semantics:**
 
 A *const-declaration* defines a c-constant.
 
-All class constants have public visibility.
+If class constants [*visibility-modifier*](#general) is omitted, `public` is assumed.
```

Comment:
Nit: Grammar. Maybe "If the visibility-modifier for a class constant is omitted, public is assumed"?